libs/wire-api/src/Wire/API/Routes/Public/Galley/Conversation.hs

```diff
@@ -898,6 +965,9 @@ type ConversationAPI =
     :<|> Named
            "update-conversation-access"
            ( Summary "Update access modes for a conversation"
+               :> MakesFederatedCall 'Galley "on-conversation-updated"
```

Comment:
This applies to any of these endpoints and any federated call that might happen and I'm picking just this one as an example. Suppose one day this endpoint does not make the federated call in Galley for the "on-conversation-updated" endpoint any more. Will the type checker tell us this is a redundant constraint? I suppose it will not.

We're in a similar problem with the `CanThrow` combinator. We have to declare all the errors an endpoint's handler can throw, but should we remove an error from the handler we get no warning the `CanThrow` combinator in the endpoint definition is now redundant.

Reply:
I just tested this, and you're right; we don't get redundant constraint warnings for this. I don't think such a thing is possible, unfortunately; GHC's support for constraint kinds is pretty lacking, and it was pretty hard to get this far.

Reply:
There is a possible solution here by way of a GHC plugin, but I think it's ridiculously overkill. Pinging @julialongtin so she knows it's possible if it ever turns out to be a problem in the future, but I don't think it's worth doing now.